### PR TITLE
fix NaN appearing when do_mac_project = 0

### PR DESCRIPTION
**Pull request by b94209002**
NaN appears in standard 2d example taylor green.  Add the following to fix this problem.

### CODE REVIEW

File: Source/NavierStokes.cpp [hunk 1]
```diff
@@ -339,6 +339,9 @@ NavierStokes::advance (Real time,
         MultiFab& S_old = get_old_data(State_Type);
 	// NOTE have_divu is now a static var in NSBase
         mac_project(time,dt,S_old,&mac_rhs,umac_n_grow,true);
+    } else { 
+	u_mac[0].FillBoundary(0, 1, geom.periodicity());
+        u_mac[1].FillBoundary(0, 1, geom.periodicity());
```
Comment (drummerdoc) [inline Source/NavierStokes.cpp:344]: You need to loop over AMREX_SPACEDIM in case you are in 3D

Comment (b94209002) [reply Source/NavierStokes.cpp:344]: yes, l am trying the other way around using create_umac_grown(umac_n_grow). I will update in a minute.

**Comment by b94209002** [comment]
I update this commit with create_umac_grown(umac_n_grow), so the loop over AMREX_SPACEDIM won't be necessary.
This change also moves create_umac_grown to protected in NavierStokesBase.H

**Comment by drummerdoc** [comment]
Yes, this is a much heavier weight function, but seems more consistent with what happens with no mac_project, in that at coarse-fine, the grow cells are fill by interpolation using coarse umac.  Looks good.

**Comment by emotheau** [comment]
Can we merge in development if everything is good?

**Comment by drummerdoc** [comment]
Look good to me

**Pull request merged**